pulumi/infra/swarm.py

```diff
@@ -0,0 +1,209 @@
+                from_port=80,
+                to_port=80,
+                protocol="tcp",
+                self=True,
+            ),
+        ] + [egress for _, egress in internal_rules]
+
+        self.security_group = aws.ec2.SecurityGroup(
+            "SwarmSecurityGroup",
+            opts=child_opts,
+            description=f"Docker Swarm security group",
+            ingress=ingress_rules,
+            egress=egress_rules,
+            vpc_id=vpc.id,
+            name=f"{DEPLOYMENT_NAME}-grapl-swarm",
+        )
+
+        self.role = aws.iam.Role(
+            f"{DEPLOYMENT_NAME}-grapl-swarm-role",
+            description="IAM role for Swarm instances",
+            assume_role_policy=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Action": "sts:AssumeRole",
+                            "Effect": "Allow",
+                            "Principal": {
+                                "Service": "ec2.amazonaws.com",
+                            },
+                        }
+                    ],
+                }
+            ),
+            opts=child_opts,
+        )
+
+        # CloudWatchAgentServerPolicy allows the Swarm instances to
+        # run the CloudWatch Agent.
+        policies._attach_CloudWatchAgentServerPolicy(role=self.role, opts=child_opts)
+        policies._attach_AmazonSSMManagedInstanceCore(role=self.role, opts=child_opts)
+        policies._attach_policy_to_ship_logs_to_cloudwatch(
+            role=self.role, log_group=log_group, opts=child_opts
+        )
+
+        self.swarm_hosted_zone = aws.route53.Zone(
+            "SwarmZone",
+            name=f"{DEPLOYMENT_NAME}.dgraph.grapl",
+            vpcs=[
+                aws.route53.ZoneVpcArgs(
+                    vpc_id=vpc.id,
+                )
+            ],
+            opts=child_opts,
+        )
+
+        self.swarm_config_bucket = Bucket(
+            logical_bucket_name="swarm-config-bucket",
+            opts=child_opts,
+        )
+        self.swarm_config_bucket.grant_read_permissions_to(self.role)
+        self.swarm_config_bucket.upload_to_bucket(SWARM_INIT_DIR)
+
+        # InstanceProfile for swarm instances
+        aws.iam.InstanceProfile(
+            "SwarmInstanceProfile",
+            opts=child_opts,
+            role=self.role.name,
+            name=f"{DEPLOYMENT_NAME}-swarm-instance-profile",
+        )
+
+    def cluster_host_port(self) -> pulumi.Output[str]:
+        return Output.concat(self.swarm_hosted_zone.name, ":9080")
+
+    def allow_connections_from(
+        self, other: aws.ec2.SecurityGroup, port_range: Ec2Port, opts: ResourceOptions
+    ) -> None:
+        descriptor = "".join(
+            [
+                "from_",
+                other._name,
+                "_to_",
+                self.security_group._name,
+                "_for_",
+                str(port_range),
```

Comment:
Might want a custom string method for this; as it is, this generates a name with content like `Ec2Port(protocol='tcp', port=9080)`; something like `TCP:9080` would be just as informative, but less verbose.

Reply:
yeah, sure

Reply:
probably both `__str__` and `__repr__` should be like `f"{protocol}:{port}"`?

Reply:
Yeah, that's more what I was thinking, too.

`egress-from-graphql-endpoint-security-group-for-tcp:9080`
is nicer than
`egress-from-graphql-endpoint-security-group-for-Ec2Port(tcp:9080)`